docs/modelserving/v1beta1/torchserve/README.md

```diff
@@ -296,6 +404,135 @@ curl -v -H "Host: ${SERVICE_HOSTNAME}" http://${INGRESS_HOST}:${INGRESS_PORT}/v2
     ```
 
+## Deploy PyTorch Model with V2 GRPC Protocol
+
+
+### Create the InferenceService
+
+For deploying the `InferenceService` with GRPC protocol you need to expose the GRPC port on InferenceService. Here **8081** is kserve GRPC port.
+
+Apply the following [mnist_grpc_v2.yaml](./mnist_grpc_v2.yaml) to create the `InferenceService`.
+
+=== "kubectl"
+```bash
+kubectl apply -f mnist_grpc_v2.yaml
+```
+
+Expected Output
+
+```bash
+$inferenceservice.serving.kserve.io/torchserve-grpc-v2 created
+```
+
+=== "Old Schema"
+
+    ```yaml
+    apiVersion: serving.kserve.io/v1beta1
+    kind: InferenceService
+    metadata:
+      name: "torchserve-grpc-v2"
+    spec:
+      predictor:
+        pytorch:
+          protocolVersion: v2
+          storageUri: gs://kfserving-examples/models/torchserve/image_classifier/v2
+          ports:
+            - containerPort: 8081
+              name: h2c
+              protocol: TCP
+    ```
+
+=== "New Schema"
+
+    ```yaml
+    apiVersion: serving.kserve.io/v1beta1
+    kind: InferenceService
+    metadata:
+      name: "torchserve-grpc-v2"
+    spec:
+      predictor:
+        model:
+          modelFormat:
+            name: pytorch
+          protocolVersion: v2
+          storageUri: gs://kfserving-examples/models/torchserve/image_classifier/v2
+          ports:
+            - containerPort: 8081
+              name: h2c
+              protocol: TCP
+    ```
+
+
+### Download kserve's grpc inference proto
+
+```bash
+mkdir -p proto/v2
+
+PROTO_FILE_PATH=https://raw.githubusercontent.com/kserve/kserve/master/python/kserve/kserve/protocol/grpc/grpc_predict_v2.proto
+
+curl -s -L ${PROTO_FILE_PATH} > ./proto/v2/grpc_predict_v2.proto
+```
+
+### Run Model Inference
+
+The first step is to [determine the ingress IP and ports](../../../get_started/first_isvc.md#4-determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`.
+
+```bash
+INPUT_PATH=./mnist_v2_grpc_tensor.json
+PROTO_FILE=proto/v2/grpc_predict_v2.proto
+SERVICE_HOSTNAME=$(kubectl get inferenceservice torchserve-grpc-v2 -o jsonpath='{.status.url}' | cut -d "/" -f 3)
+```
+
+### Make grpc call
```

Comment:
Make gRPC call is the same step as run model inference, we can merge the two titles